### ISSUE TITLE
Problem with |= applied to a set passed as an argument

### ISSUE THREAD

**Issue by smontanaro** [PyCQA/pylint#774]
(I'm using pylint 1.4.2 from the Anaconda distribution, and Python 2.7.11.)

[junk.txt](https://github.com/PyCQA/pylint/files/86648/junk.txt)
Consider the attached code, with two versions of the same function, find_problems1 and find_problems2.
When I pylint that script, find_problems2 triggers a message:

```
junk.py:16: [W0613(unused-argument), find_problems2] Unused argument 'prob_dates'
```

No such message is emitted for find_problems1.

Running the code as a program (so the two dis.dis functions are calls), yields two slightly different versions of the generated byte code. For find_problems1:

```
  14         62 LOAD_FAST                3 (prob_dates)
             65 LOAD_ATTR                6 (update)
             68 LOAD_GLOBAL              7 (compare_prices)
             71 LOAD_CONST               3 ('E:%s')
             74 LOAD_FAST                5 (sym)
             77 BINARY_MODULO      
             78 LOAD_FAST                0 (cx1)
             81 LOAD_FAST                1 (cx2)
             84 LOAD_FAST                2 (cx3)
             87 CALL_FUNCTION            4
             90 CALL_FUNCTION            1
```
For find_problems2:

```
  20         62 LOAD_FAST                3 (prob_dates)
             65 LOAD_GLOBAL              6 (compare_prices)
             68 LOAD_CONST               3 ('E:%s')
             71 LOAD_FAST                5 (sym)
             74 BINARY_MODULO       
             75 LOAD_FAST                0 (cx1)
             78 LOAD_FAST                1 (cx2)
             81 LOAD_FAST                2 (cx3)
             84 CALL_FUNCTION            4
             87 INPLACE_OR          
             88 STORE_FAST               3 (prob_dates)
```

I believe the STORE_FAST opcode at offset 88 is causing the problem. Since formal arguments are also local variables, it seems the LOAD_FAST at offset 62 should satisfy pylint that the argument is, in fact, not ignored, despite the fact that the STORE_FAST opcode overwrites it. The INPLACE_OR opcode should be a very strong hint that prob_dates is mutable, and that we've referenced it (offset 62) and modified it (offset 87).

**Comment by PCManticore** [comment]
I can't reproduce and this seems to be fixed in pylint 1.5.X. Could you give it a try and tell me if it works for you? I also added a test for checking this (https://github.com/PyCQA/pylint/commit/adeeb296082f4543fb2958577ced2459403bf364)

**Comment by smontanaro** [comment]
> I can't reproduce and this seems to be fixed in pylint 1.5.X. Could you give
> it a try and tell me if it works for you? I also added a test for checking
> this (adeeb29)

Github won't allow me to add a comment for some reason. Trying email
and CC'ing Claudiu (just in case)...

I confirmed that this is fixed with pylint 1.5.4, the latest version
available with Anaconda.

Skip
